Sources/FluentKit/Query/QueryBuilder+Distinct.swift

```diff
@@ -0,0 +1,33 @@
+extension QueryBuilder {
+    public func distinctCount<Field>(on key: KeyPath<Model, Field>) -> EventLoopFuture<Int>
+        where Field: FieldRepresentable
+    {
+        let copy = self.copy()
+        let fieldKey = Model.key(for: key)
+        copy.query.fields = [.aggregate(.fields(method: .count, fields: [
+                .function(.distinct, fields: [.field(path: [fieldKey], schema: Model.schema, alias: nil)])
+            ]))
+        ]
+        return copy.first().flatMapThrowing { res in
+            guard let res = res else {
+                throw FluentError.noResults
+            }
+            return try res._$id.cachedOutput!.decode("fluentAggregate", as: Int.self)
+        }
+    }
+    
+    public func distinct<Result>(on key: KeyPath<Model, Field<Result>>) -> EventLoopFuture<[Result]> where Result: Decodable {
```

Comment:
Hmm I'm not sure about the naming here. What about these ideas:

```swift
Planet.query(...).all(\.$name)
// or 
Planet.query(...).all(distinct: \.$name)
```

Reply:
I like them, but I'd personally suggest
```swift
Planet.query(...).all(distinctOn: \.$name)
```
(I like the `on` part, makes it read more nice IMO)